simple-weather/simple-weather/Controller/ViewController.swift

```diff
@@ -49,10 +49,22 @@ class ViewController: UIViewController, CLLocationManagerDelegate {
             switch result {
             case .success(let currentWeather):
                 DispatchQueue.main.async {
-                    self.locationLabel.text = ("\(currentWeather.cityName), \(currentWeather.countryType)")
-                    self.tempLabel.text = "\(String(Int((currentWeather.currentTemp - 273.15))))°"
-                    self.weatherLabel.text = currentWeather.weatherType
-                    self.weatherImage.image = UIImage(named: currentWeather.weatherType)
+                    self.locationLabel.text = currentWeather.location
+                    self.tempLabel.text = NSString(format: "%.2f°C", currentWeather.temperature) as String
+                    self.weatherLabel.text = currentWeather.weather
+                    self.weatherImage.image = currentWeather.image
+                }
+            case .failure(let error):
+                print(error.localizedDescription)
+            }
+        }
+        
+        APIService().getDataForCell(coordinates: locValue) { (result) in
+            switch result {
+            case .success(let forecastWeather):
+                DispatchQueue.main.async {
+                    self.forecastWeatherArray = forecastWeather
```

Comment:
Currently `self.forecastWeatherArray = forecastWeather` sets all items to `forecastWeatherArray` which is used to display cells in table view. Before setting `forecastWeatherArray` value, You will have to filter out objects with same value of `weekDay`, and set array with filtered data (_eg._ `self.forecastWeatherArray = filteredForecast`) You will have to iterate, use `weekDay` as value by which to filter. Do not spend more than 3 hours on this step.